golang/btcspv/validate_spv.go

```diff
@@ -184,3 +184,32 @@ func ValidateHeaderChain(headers []byte) (sdk.Uint, error) {
 	}
 	return totalDifficulty, nil
 }
+
+func (s SPVProof) Validate() (bool, error) {
+	// Calculate the Tx ID and compare it to the one in SPVProof
+	txid := CalculateTxID(s.Version, s.Vin, s.Vout, s.Locktime)
+	if bytes.Compare(txid, s.TxID) != 0 {
```

Comment:
Need to use the LE txid here.
Also try casting the txid to a bytearray to solve your type issue:

`if !bytes.Equal(txid, []byte(s.TxIDLE))`

Reply:
you probably actually need to do 
`if !bytes.Equal(txid, s.TxIDLE[:])`